src/go/rpk/pkg/cli/cmd/redpanda/start_test.go

```diff
@@ -1354,22 +1354,29 @@ func TestStartCommand(t *testing.T) {
 			"--another-arbitrary-seastar-flag", "",
 		},
 	}, {
-		name: "it should allow arbitrary flags after '--'",
+		name: "it should fail if an unrecognized flag is passed",
 		args: []string{
 			"--install-dir", "/var/lib/redpanda",
 			"--",
 			"--i-just-made-this-on-the-spot", "nice",
 		},
-		postCheck: func(
-			_ afero.Fs,
-			rpArgs *rp.RedpandaArgs,
-			st *testing.T,
-		) {
-			expected := []string{
-				"--i-just-made-this-on-the-spot", "nice",
+		before: func(fs afero.Fs) error {
+			os.Args = append(
+				os.Args,
+				"--something-weird",
+				"--kfaka-ardd", "192.168.54.19",
+			)
+			return nil
+		},
+		after: func() {
+			for i, a := range os.Args {
+				if a == "--set" {
```

Comment:
Why is this trimming at the first `--set`? Offhand I would expect this to trim at `--something-weird`.

Reply:
Aaaand that's why copypasting is bad.